rcgen/src/lib.rs

```diff
@@ -727,6 +726,48 @@ impl CertificateParams {
 			..Default::default()
 		})
 	}
+	/// Generate a new certificate from the given parameters, signed by the provided issuer.
+	///
+	/// The returned certificate will have its issuer field set to the subject of the
+	/// provided `issuer`, and the authority key identifier extension will be populated using
+	/// the subject public key of `issuer`. It will be signed by `issuer_key`.
+	///
+	/// Note that no validation of the `issuer` certificate is performed. Rcgen will not require
+	/// the certificate to be a CA certificate, or have key usage extensions that allow signing.
+	///
+	/// The returned [`Certificate`] may be serialized using [`Certificate::der`] and
+	/// [`Certificate::pem`].
+	pub fn sign(
+		self,
+		key_pair: &KeyPair,
+		issuer: &Certificate,
+		issuer_key: &KeyPair,
+	) -> Result<Certificate, Error> {
+		let subject_public_key_info = key_pair.public_key_der();
+		let der = self.serialize_der_with_signer(
+			key_pair,
+			issuer_key,
+			&issuer.params.distinguished_name,
+		)?;
+		Ok(Certificate {
+			params: self,
+			subject_public_key_info,
+			der,
+		})
+	}
+	/// Generates a new self-signed certificate from the given parameters.
+	///
+	/// The returned [`Certificate`] may be serialized using [`Certificate::der`] and
+	/// [`Certificate::pem`].
+	pub fn sign_self(self, key_pair: &KeyPair) -> Result<Certificate, Error> {
```

Comment:
I think `sign_self_signed` is a better name for this. IMO it's a good idea to add both a `sign` at the start and one at the end: the first one to indicate it's analogous with the `sign` function, the second because of the established `self-signed` term. The two occurrences of `sign` are weird indeed, yes, but I think it's the least weird name. But agreeing that every choice is going to be weird in some fashion.